thirdparty/nanosvg/nanosvg.h

```diff
@@ -2732,14 +2767,69 @@ static void nsvg__endElement(void* ud, const char* el)
 		p->pathFlag = 0;
 	} else if (strcmp(el, "defs") == 0) {
 		p->defsFlag = 0;
+	} else if (strcmp(el, "style") == 0) {
+		p->styleFlag = 0;
 	}
 }
 
+static char *nsvg__strndup(const char *s, size_t n)
+{
+	char *result;
+	size_t len = strlen(s);
+
+	if (n < len)
+		len = n;
+
+	result = (char *)malloc(len + 1);
+	if (!result)
+		return 0;
+
+	result[len] = '\0';
+	return (char *)memcpy(result, s, len);
+}
```

Comment:
That's the typical minefield for security vulnerabilities. Did you ensure that it can't do out of bounds reads or similar?
It would be better IMO to use existing C standard methods instead of reimplementing something that's coming in C23.